.github/workflows/ci.yml

```diff
@@ -0,0 +1,206 @@
+name: CI
+
+on: [pull_request, push]
+
+jobs:
+  format:
+    name: Rustfmt
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          components: rustfmt
+          profile: minimal
+          override: true
+
+      - name: Run rustfmt
+        uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all -- --check
+
+  clippy:
+    name: Clippy
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install packages
+        run: |
+          sudo apt-get update
+          sudo apt-get -yq --no-install-suggests --no-install-recommends install libasound2-dev libudev-dev
+
+      - name: Install rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          components: clippy
+          profile: minimal
+          override: true
+
+      - name: Run clippy
+        uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: -- -D warnings
+
+  assets:
+    name: Check assets
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Checkout assets
+        uses: actions/checkout@v2
+        with:
+          repository: ozkriff/zemeroth_assets
+          path: assets
+
+      - name: Check
+        run: |
+          SRC=$(grep "const ASSETS_HASHSUM" src/main.rs | cut -d\" -f 2)
+          ASSETS=$(cat assets/.checksum.md5)
+
+          echo "Source: $SRC"
+          echo "Assets: $ASSETS"
+          if [ "$SRC" != "$ASSETS" ]; then
+            false
+          fi
+
+  build:
+    name: ${{ matrix.build }}
+    runs-on: ${{ matrix.os }}
+
+    # The build matrix does not yet support 'allow failures' at job level.
```

Comment:
Hmm.. Nightly failures still mark the whole commit as "not successful".

![image](https://user-images.githubusercontent.com/662976/72226820-53809380-35a6-11ea-921a-7170bc6d5299.png)

Is it possible to do something about it?

Reply:
https://github.community/t5/GitHub-Actions/continue-on-error-allow-failure-UI-indication/td-p/37033 :(

Reply:
Yes, you can disable `fail-fast` in the build matrix.

Reply:
Yes, you can configure the required status checks under `Settings > Branches`
![image](https://user-images.githubusercontent.com/2128532/72227172-a6a01980-3599-11ea-9abc-89238610e92e.png)

Reply:
Yeah, I've already configured this and I'm able to merge the PR. It just troubles me that the commit is marked as ❌ - it feels somewhat wrong.

Reply:
Ok, i've tested it https://github.com/nickelc/zemeroth/pull/1 and `continue-on-error` for nightly is the only way to keep it green at the moment.

Reply:
Could you please add it to this PR?